test/test_client.py

```diff
@@ -505,6 +505,15 @@ def test_client_options(self):
         self.assertIsInstance(c.options.retry_writes, bool)
         self.assertIsInstance(c.options.retry_reads, bool)
 
+    def test_session_validation(self):
+        c = MongoClient(connect=False)
+
+        def try_improper_session():
+            with c._tmp_session({}):  # type:ignore
```

Comment:
Instead of testing _tmp_session we should call a public method, like insert_one.

Reply:
Done